crates/wasmtime/src/runtime/vm/sys/unix/mmap.rs

```diff
@@ -134,8 +138,10 @@ impl Mmap {
     }
 
     #[inline]
-    pub fn len(&self) -> usize {
-        self.memory.as_ptr().len()
+    pub fn len(&self) -> HostAlignedByteCount {
+        // SAFETY: When self.memory was created it was always aligned to a page
+        // boundary.
+        unsafe { HostAlignedByteCount::new_unchecked(self.memory.as_ptr().len()) }
```

Comment:
Reviewing some other code elsewhere, I think that the `from_file` constructor may cause problems here because file lengths aren't always page-aligned. 

To help catch this mind adding a `debug_assert!` to `new_unchecked`? That I think should make the test suite panic

Reply:
This ended up becoming #9639.